src/test/java/io/appium/java_client/touch/FailsWithMatcher.java

```diff
@@ -0,0 +1,43 @@
+package io.appium.java_client.touch;
+
+import static org.hamcrest.core.AllOf.allOf;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+public final class FailsWithMatcher<E extends Throwable> extends TypeSafeMatcher<Runnable> {
+
+    private final Matcher<? super E> matcher;
+
+    private FailsWithMatcher(final Matcher<? super E> matcher) {
+        this.matcher = matcher;
+    }
+
+    public static <E extends Throwable> Matcher<Runnable> failsWith(
+            final Class<E> throwableType) {
+        return new FailsWithMatcher<>(instanceOf(throwableType));
+    }
+
+    public static <E extends Throwable> Matcher<Runnable> failsWith(
+            final Class<E> throwableType, final Matcher<? super E> throwableMatcher) {
+        return new FailsWithMatcher<>(allOf(instanceOf(throwableType), throwableMatcher));
```

Comment:
Also compilation error here

Reply:
??? ок

Reply:
@SrinivasanTarget 
I can't reproduce it even after removal of the project from the disc and cloning it again. 
Could you try again? If you are facing the same issue could you provide some text of the error.

Reply:
@SrinivasanTarget even Travis could compile the project

Reply:
Yes travis wasn't able to catch it. Only IDE throws it on file level but still project builds fine. `no instance of type variable exists so that E conforms to capture of ? super....` is error.

Reply:
looks fine in latest IDE

Reply:
Do you use Eclipse?

Reply:
Intellij IDEA 2017.1.5 throws this error but no errors in Intellij IDEA 2017.2